### PR TITLE
Adds Apollos.json to help improve upgrade tool.

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

This PR creates an `apollos.json` file. This file only exists so that the upgrade tool has something to work with when assuming the user's current version and current environment. With this file, we can simplify the code required to complete the process of minimizing the args that you need to pass to the upgrade tool. 

When you run the upgrade tool, it will look at this file to determine it's context (api or client) and then look at the version to find the "current version". When the upgrade tool is run, this file will automatically be updated to reference the new current version (the end user will never have to touch this file). 

I also added steps to the release process so that this file stays in sync with the current package version.

## TODO:

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apolloschurchapp/apollos.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "version": "0.8.5",
+    "environment": "client"
```
Comment (conrad-vanl) [inline packages/apolloschurchapp/apollos.json]: how is `environment` used on the upgrade tool? should this name be more explicit? (like `apolloschurchapp` or something that directly references what it is?) I'm just trying to think future proof....

Comment (vinnyjth) [reply packages/apolloschurchapp/apollos.json]: It's flag that we currently pass. It's either `api` or `client`, and so this is a direct translations of that flag. it drives whether the `diff` used to upgrade is the client diff or the api diff. 

 I think this is okay as it stands right now :)

Comment (vinnyjth) [reply packages/apolloschurchapp/apollos.json]: If you look at the upgrade tool branches, I use the `api`/`client` language very consistently. 
https://github.com/ApollosProject/apollos-upgrade/branches

And here's how we specify the "platform". Maybe I should rename this key to platform....
https://github.com/ApollosProject/apollos-upgrade/blob/master/src/index.js#L27

**Comment by IsaacHardy** [comment]
Assuming this works as intended, I would run `apollos-upgrade upgrade --projectName newspringchurchapp --packageName com.newspringchurchapp` and we'd setup the script to fetch the apollos.json from CORE and read this local file to determine current version. 

Is this correct?

**Comment by vinnyjth** [comment]
> Assuming this works as intended, 
You bet it works as intended ;) 

>I would run apollos-upgrade upgrade --projectName newspringchurchapp --packageName com.newspringchurchapp and we'd setup the script to fetch the apollos.json from CORE and read this local file to determine current version.

This is almost correct. We read the current version from the remote lerna file (which you already did) and then read the local file for the current version.

**Comment by conrad-vanl** [comment]
My only worry is that will it matter if we have multiple "client" templates? like a react-native app, a conference app, a web site (down the road)

**Comment by vinnyjth** [comment]
> My only worry is that will it matter if we have multiple "client" templates? like a react-native app, a conference app, a web site (down the road)

That's a good callout 🤔 In that case, we will have to reorg more than once place (also have to upgrade the diff repo to keep track of the different clients). For now I think it's safe to keep it as simple as `client/api`

**Pull request merged**